Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -732,6 +733,15 @@ class Gx2Fitter {
     BoundMatrix aMatrix = BoundMatrix::Zero();
     BoundVector bVector = BoundVector::Zero();
 
+    // We need to create a temporary track container. We create several times a
+    // new track and delete it after updating the parameters. However, if we
+    // would work on the externally provided track container, it would be
+    // difficult to remove the correct track, if it contains more than one.
+    Acts::VectorTrackContainer trackContainerTempBackend;
+    Acts::VectorMultiTrajectory trajectoryTempBackend;
+    TrackContainer trackContainerTemp{trackContainerTempBackend,
+                                      trajectoryTempBackend};
```

Comment:
I wonder if it would be easier to have input and output containers which are both provided by the user instead of having temporary containers.

In case of the CKF we use a temporary track state container which is provided directly by the user. Hardcoding vector containers here works but might not be what the user expects.

Reply:
The idea was to have a container that:
- is decoupled from the user. The `trackContainerTemp` is only there for performing the actual fitting, therefore, the user shouldn't see anything of it.
- is always the same, which makes it more predictable.

Without performance studies, I don't see any benefit to expose this container to the user (use an user-provided container). Best case, the user gets an empty container back, worst case (what led us here), the user gets a faulty track.

Reply:
makes sense 👍